### PR TITLE
Feature/75 global header fe

### PR DESCRIPTION
**Pull request by nicklee**
#75 

Adds FE for the main navigation

Screenshots:

Screenshot 2022-10-06 at 12 33 40
Screenshot 2022-10-06 at 12 33 35
Screenshot 2022-10-06 at 12 33 22
Screenshot 2022-10-06 at 12 33 19
Screenshot 2022-10-06 at 12 33 17

### CODE REVIEW

File: wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<a class="nav-cta" href="{{ cta.url }}" title="{{ cta.text }}">
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html:1]: question: what's the title attribute needed for?

File: wagtailio/static/js/components/desktop-sub-menu.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      );
+      this.activeClass = 'active';
+      this.bindEventListeners();
+  }
+
+  bindEventListeners() {
+      this.node.addEventListener('click', (e) => {
+          e.preventDefault();
+          /* eslint-disable no-console */
+        console.log('test')
```
Comment (chris-lawton) [inline wagtailio/static/js/components/desktop-sub-menu.js:24]: Is this still needed?

File: wagtailio/static/sass/components/_nav-cta.scss [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            padding-top: 100px;
+        }
+    }
+
+    &__heading {
+        font-size: 45px;
+        line-height: 1.1;
+        margin-bottom: 10px;
+        padding-right: 60px;
+        transition: color $transition;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_nav-cta.scss:57]: question: is `color` set somewhere else? Can't see it in the list of properties listed here

File: wagtailio/static/sass/components/_primary-nav.scss [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        #{$root}__container {
+            padding: 0 20px;
+
+            @include media-query(medium) {
+                padding: 0 50px;
+            }
+        }
+
+
+
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_primary-nav.scss:121]:
```suggestion
```

File: wagtailio/static/sass/components/_sub-nav.scss [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    $root: &;
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+    width: 100%;
+    height: calc(100% - #{$header-mobile-height});
+    position: absolute;
+    list-style: none;
+    top: 0;
+    left: 0;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_sub-nav.scss:12]: Thought: surprised the linter didn't moan about using `inset` here

File: wagtailio/static/sass/components/_sub-nav.scss [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    border-radius: $rounded-xl;
+    // Extra border for high-contrast mode users.
+    border: 1px solid transparent;
+    opacity: 0;
+    visibility: hidden;
+    overflow-y: scroll;
+    overflow-x: hidden;
+    // visible to invisible
+    transition: visibility 0ms ease-out 250ms, opacity 250ms ease-out 0ms;
+    box-shadow: 0 0 30px rgba($color--black, 0.05) ;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_sub-nav.scss:24]:
```suggestion
    box-shadow: 0 0 30px rgba($color--black, 0.05);
```

File: wagtailio/static/sass/components/_sub-nav.scss [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+        display: none;
+        align-items: flex-start;
+        width: 100%;
+        height: auto;
+        top: 120px;
+        background-color: $color--white;
+        padding: 0;
+        opacity: 1;
+        overflow: hidden;
+        visibility: visible;
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_sub-nav.scss:41]: Question: are all the rules under `display: none` needed if the element is hidden?

File: wagtailio/static/sass/layout/_header.scss [hunk 8]
```diff
@@ -8,43 +8,75 @@
...
-    @include media-query(large) {
-        padding: 35px 50px;
+    &__actions {
+        display: flex;
+        align-items: center;
+        position: relative;
+        top: -7px;
+
+        @include media-query(medium) {
+            top: -7px;
```
Comment (chris-lawton) [inline wagtailio/static/sass/layout/_header.scss:22]: Question: can this be removed as it's the same as the line 19?

**Comment by chris-lawton** [comment]
Thanks @nicklee - left some questions/suggestions

File: wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html (hunk 1 above)
Comment (nicklee) [reply wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html:1]: removed :)

File: wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    <svg class="nav-cta__icon">
```
[the remark this answers is not in the document]
Comment (nicklee) [reply wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html:2]: Done

File: wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    <div class="nav-cta__content">
+        <h3 class="nav-cta__heading">{{ cta.text }}</h3>
+        {% if cta.sub_heading %}
+            <h4 class="nav-cta__subheading">{{ cta.sub_heading }}</h4>
+        {% endif %}
+        {% if cta.description %}
+            <p class="nav-cta__description">{{ cta.description }}</p>
+        {% endif %}
+    </div>
+    <svg class="nav-cta__arrow">
```
[the remark this answers is not in the document]
Comment (nicklee) [reply wagtailio/project_styleguide/templates/patterns/components/nav-cta/nav-cta.html:14]: Done

File: wagtailio/static/js/components/desktop-close-menus.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+import DesktopSubMenu from './desktop-sub-menu';
+
+class DesktopCloseMenus {
+    constructor() {
+        this.desktopSubMenus = document.querySelectorAll(
+            DesktopSubMenu.selector(),
+        );
+        this.allPrimaryNavs = document.querySelectorAll(
+            '[data-desktop-menu] [data-primary-nav]',
```
[the remark this answers is not in the document]
Comment (nicklee) [reply wagtailio/static/js/components/desktop-close-menus.js:12]: just the one, nice one, updated

File: wagtailio/static/js/components/desktop-sub-menu.js (hunk 2 above)
Comment (nicklee) [reply wagtailio/static/js/components/desktop-sub-menu.js:24]: oops

File: wagtailio/static/sass/components/_nav-cta.scss (hunk 3 above)
Comment (nicklee) [reply wagtailio/static/sass/components/_nav-cta.scss:56]: it's inherited by the main .nav-cta and a hover applied on the same selector

File: wagtailio/static/sass/layout/_header.scss (hunk 8 above)
Comment (nicklee) [reply wagtailio/static/sass/layout/_header.scss:22]: oops, sure can, thanks

File: wagtailio/static/sass/components/_sub-nav.scss (hunk 7 above)
Comment (nicklee) [reply wagtailio/static/sass/components/_sub-nav.scss:41]: It's togglable, so these are applied when `.active` is

File: wagtailio/static/sass/components/_sub-nav.scss (hunk 5 above)
Comment (nicklee) [reply wagtailio/static/sass/components/_sub-nav.scss]: updated all left to inset for consistency

**Pull request merged**